nat-lab/tests/conftest.py

```diff
@@ -232,9 +233,75 @@ async def _copy_vm_binaries_if_needed(items):
                 return
 
 
+def save_dmesg_from_host(suffix):
+    try:
+        result = subprocess.run(
+            ["sudo", "dmesg", "-d", "-T"],
+            capture_output=True,
+            text=True,
+            check=True,
+        ).stdout
+    except subprocess.CalledProcessError as e:
+        print(f"Error executing dmesg: {e}")
+        return
+
+    if result:
+        with open(
+            os.path.join("logs", f"dmesg-{suffix}.txt"), "w", encoding="utf-8"
+        ) as f:
+            f.write(result)
+
+
+def save_audit_log(suffix):
+    try:
+        source_path = "/var/log/audit/audit.log"
+        if os.path.exists(source_path):
+            shutil.copy2(source_path, f"logs/audit_{suffix}.log")
+        else:
+            print(f"The audit file {source_path} does not exist.")
+    except Exception as e:  # pylint: disable=broad-exception-caught
+        print(f"An error occurred when processing audit log: {e}")
+
+
+async def _save_macos_logs(conn, suffix):
+    try:
+        dmesg_proc = await conn.create_process(["dmesg"]).execute()
+        with open(
+            os.path.join("logs", f"dmesg-macos-{suffix}.txt"), "w", encoding="utf-8"
+        ) as f:
+            f.write(dmesg_proc.get_stdout())
+    except ProcessExecError as e:
+        print(f"Failed to collect dmesg logs {e}")
+
+
+async def collect_kernel_logs(items, suffix):
+    if os.environ.get("NATLAB_SAVE_LOGS") is None:
+        return
+
+    is_ci = os.environ.get("CUSTOM_ENV_GITLAB_CI") is not None and os.environ.get(
+        "CUSTOM_ENV_GITLAB_CI"
+    )
+
+    log_dir = "logs"
+    os.makedirs(log_dir, exist_ok=True)
+
+    save_dmesg_from_host(suffix)
+    save_audit_log(suffix)
```

Comment:
I think the naming is a bit inconsistent, both are saving logs from the machine, so maybe just `save_dmesg_log` and `save_audit_log` would be enough? :thinking:

Reply:
I'll change `save_audit_log_from_host`, adding `from_host` suffix makes it more clear where the log is comming from